### PR TITLE
Simple improvements

### PR DESCRIPTION
**Pull request by pferreir**
* De-duplication of identities in list of participants
 * Adding users from the management area now bypasses pending state

### CODE REVIEW

File: indico/modules/events/registration/__init__.py [hunk 1]
```diff
@@ -76,8 +76,12 @@ def _inject_event_header(event, **kwargs):
                                           RegistrationForm.publish_registrations_enabled,
                                           _join=Registration.registration_form)
 
+    linked_participants = {reg.user: reg for reg in registrations if reg.user is not None}
+    non_linked_participants = set(reg for reg in registrations if reg.user is None)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/__init__.py]: set literals are a nice thing :p

File: indico/modules/events/registration/__init__.py [hunk 2]
```diff
@@ -76,8 +76,12 @@ def _inject_event_header(event, **kwargs):
...
+    participants = sorted(set(linked_participants.iterkeys()) | non_linked_participants, key=lambda p: p.id)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/__init__.py]: `linked_participants.viewkeys()` already gives you a set-like object

`key=attrgetter('id')`

File: indico/modules/events/registration/models/registrations.py [hunk 3]
```diff
@@ -342,7 +342,7 @@ def render_price(self):
         currency = event_payment_settings.get(self.registration_form.event, 'currency')
         return '{} {}'.format(self.price, currency)
 
-    def update_state(self, approved=None, paid=None, rejected=None):
+    def update_state(self, approved=None, paid=None, rejected=None, manager=False):
```
Comment (ThiefMaster) [inline indico/modules/events/registration/models/registrations.py]: I'd call it `management` (other functions in the module with similar management-sepcific behavior all use `management` for the argument)

File: indico/modules/events/registration/__init__.py (hunk 1 above)
Comment (pferreir) [inline indico/modules/events/registration/__init__.py]: old habits die hard.

File: indico/modules/events/registration/models/registrations.py (hunk 3 above)
Comment (pferreir) [inline indico/modules/events/registration/models/registrations.py]: You're right, but it sounds a bit weird.

**Pull request merged**